### PR TITLE
Pipelines notifications

### PR DESCRIPTION
**Pull request by cheikhgwane**
We introduce a new, fine-grained way of configuring pipelines notifications : add possibility to specify for each users (recipients) a notification level. We also changed the pipeline page view and introduced a tabular navigation.

## Changes

Please list / describe the changes in the codebase for the reviewer(s).

- Introduce a TabLayout
- Use a tab navigation system (same as dataset page) and split pipeline page into : General, Runs and Scheduling & Notifications section.
- Add a component to handle list/create/edit of a recipient.
- Allow user to override notification for a run (mute)

## Screenshots / screencast

https://github.com/user-attachments/assets/350bf520-097d-40e9-b92b-717a0280fd7a

Screenshot 2024-11-12 at 16 18 05

### CODE REVIEW

**Comment by qgerome** [comment]
> Also, apply a negative x margin for the recipients table. It's going to look nicer Screenshot 2024-11-13 at 16 08 07

I didn't see the change in the latest commits. The goal is to have the gray background colour to go from each border left & right of the Block.

**Review by qgerome — CHANGES_REQUESTED** [review]
There are lot of queries that are executed on the frontend even though it should not since it's already rendered on the backend. By having a quick look at the getServerSideProps of all the pages, I see that it still uses `WorkspaceLayout.prefetch` even though it's not used in the page. You need to implement this function in your layout. Check in the end the queries in the network panel.

**Comment by cheikhgwane** [comment]
> There are lot of queries that are executed on the frontend even though it should not since it's already rendered on the backend. By having a quick look at the getServerSideProps of all the pages, I see that it still uses `WorkspaceLayout.prefetch` even though it's not used in the page. You need to implement this function in your layout. Check in the end the queries in the network panel.

By trying to debug the issues, I've noticed that in local (for the pipeline  details page) some queries are run twice or the previous page query (e.g : navigate to pipelines/ then go to pipeline/[pipelineCode]) is also run on the page. Still investigating

File: src/pages/workspaces/[workspaceSlug]/pipelines/[pipelineCode]/notifications.tsx
Comment (qgerome) [inline src/pages/workspaces/[workspaceSlug]/pipelines/[pipelineCode]/notifications.tsx:1]: This page still send a network request for the `WorkspacePipelineNotificationsPage` query (even after updating the branch on the Apollo fix)

File: src/workspaces/features/WorkspaceMemberPicker/WorkspaceMemberPicker.tsx [hunk 1]
```diff
@@ -28,8 +30,10 @@ const WorkspaceMemberPicker = (props: WorkspaceMemberPickerProps) => {
...
     required = false,
     withPortal = false,
+    multiple = false,
     onChange,
-    placeholder = t("Select recipients"),
+    exclude = [],
+    placeholder = t("Select member"),
   } = props;
 
   const { data, loading } = useQuery<WorkspaceMemberPickerQuery>(
```
Comment (qgerome) [inline src/workspaces/features/WorkspaceMemberPicker/WorkspaceMemberPicker.tsx:39]: This component triggers two queries on mount. It's better to use a LazyQuery that would fetch the list only when the user open the select.

**Review by qgerome — CHANGES_REQUESTED** [review]

File: src/pages/workspaces/[workspaceSlug]/pipelines/[pipelineCode]/notifications.tsx
Comment (cheikhgwane) [reply src/pages/workspaces/[workspaceSlug]/pipelines/[pipelineCode]/notifications.tsx:1]: Yes, I've noticed that, currently on it, thx.

**Pull request merged**